app/services/customers/metadata/update_service.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Customers
+  module Metadata
+    class UpdateService < BaseService
+      def initialize(customer:)
+        @customer = customer
+        super
+      end
+
+      def call(params:)
+        created_metadata_ids = []
+
+        hash_metadata = params.map { |m| m.to_h.deep_symbolize_keys }
+        hash_metadata.each do |payload_metadata|
+          metadata = customer.metadata.find_by(id: payload_metadata[:id])
+
+          if metadata
+            metadata.update!(payload_metadata)
+
+            next
+          end
+
+          created_metadata = create_metadata(payload_metadata)
+          created_metadata_ids.push(created_metadata.id)
+        end
+
+        # NOTE: Delete metadata that are no more linked to the customer
+        sanitize_charges(hash_metadata, created_metadata_ids)
```

Comment:
Shouldn't we name this one `sanitize_metadata`?

Reply:
Oh, thanks for catching!